### DISCUSSION TITLE
Welcome to booking-microservices-nestjs Discussions!

### DISCUSSION THREAD

**Discussion by meysamhadeli** [meysamhadeli/booking-microservices-nestjs#5] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by viniciusdeliz** [comment]
@meysamhadeli great job! Thank you for publishing, maintaining and keeping this project available publicly. I have a question: why the project does not use the [NestJS Microservices](https://docs.nestjs.com/microservices/basics) native feature?

**Reply by meysamhadeli** [reply]
@viniciusdeliz Your welcome :) because I was looking some advance features of Rabbitmq like handle exchange and Nest Rabbitmq just handle it in behind the scenes and we can't configure it by ourself and also some integration for tracing and monitoring with open telemetry and built in nest microservices was not very flexible for implement those, and I decided to implement by myself to have more flexibility and easy to maintain by myself. maybe nest microservices in next versions it would be more match with advance features in microservices.
